s64da_benchmark_toolkit/correctness.py

```diff
@@ -2,24 +2,68 @@
 import logging
 import os
 
+from enum import Enum
+
 import numpy as np
 import pandas as pd
 
 from natsort import index_natsorted, order_by_index
-from pandas.io.formats.style import Styler
 
 LOG = logging.getLogger()
 
+
+class CorrectnessResult:
+    def __init__(self, status, detail=None, truth=[], result=[]):
+        self.status = status
+        self.detail = detail
+        self.truth = truth
+        self.result = result
+
+    @classmethod
+    def make_mismatch_result(cls, detail, truth, result):
+        return cls('MISMATCH', detail=detail, truth=truth, result=result)
+
+    @classmethod
+    def make_ok_result(cls):
+        return cls('OK')
+
+    @property
+    def is_ok(self):
+        return self.status == 'OK'
+
+    @property
+    def is_mismatch(self):
+        return self.status == 'MISMATCH'
+
+    def to_html(self):
+        if self.is_ok:
+            return self.status
+
+        # HTML here, since it'll be used for reporting to HTML
+        truth_html = self.truth.to_html()
+        result_html = self.result.to_html()
+        return f'{value.status}<br /><div>{truth_html}</div><br /><div>{result_html}</div>'
```

Comment:
`{value.status}` should now be `{self.status}`

Reply:
Good one. Fixed it. Probably needs a test next time haha